modules/javafx.graphics/src/test/java/test/com/sun/javafx/iio/common/ImageToolsTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.com.sun.javafx.iio.common;
+
+import com.sun.javafx.iio.common.ImageTools;
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+import java.util.Random;
+
+public class ImageToolsTest {
+    private static final int RANDOM_SEED = 1;  // A random seed
+    private static final int MIN_SIZE = 1;  // Smallest dimension to generate
+    private static final int MAX_SIZE = 1000;  // Largest dimension to generate
+    private static final int RANGE = MAX_SIZE - MIN_SIZE;  // Generate dimension range
+    private static final int RANDOM_COUNT = 1000;  // Number of random source and target dimensions to generate and verify
+
+    @Test
+    public void testIfComputeDimensionsReturnsValuesInCorrectRangeWhenAspectRatioIsPreserved() {
+        Random rnd = new Random(RANDOM_SEED);
```

Comment:
I still want to see some tests with explicit values, including a few with one dimension being 0. You might create a utility method that takes the 4 values, which can then be called with a few carefully chosen values and then also called in the loop with random values. Especially in light of the comment I made on the fix itself.

As an example, the following will produce different results because the first will (after your fix) be rounded:

```
sx = 98, sy = 97, mx = 40, my= 50 -- result: x = 40, y = 40 (with your fix)
sx = 98, sy = 97, mx = 40, my= 0 -- result: x = 40, y = 39 (with or without your fix)
```

Reply:
I update the test with your suggestions, and included sufficient cases to cover all branches of the entire function.